tests/core/test_rtdb.py

```diff
@@ -1,21 +1,351 @@
+    """Tests the get_all_table_object_dict method of the RedisDB class with json decode error.
+
+    This test checks if the get_all_table_object_dict method can get all dict values from a table
+    on redis server successfully.
+
+    Args:
+        my_redis_client (Callable): Temporary redis client provided by pytest-redis's redisdb
+          fixture.
+        rtdb_connect (RuntimeDatabaseBase): Fixture for Redis runtime database.
+        data_config (dict): Fixture for test data configurations.
+
+    Returns:
+        None
+    """
+    my_redis_client.hset(data_config['table'], data_config['obj'], data_config['dict'])
+    assert rtdb_connect.get_all_table_objects_dict(data_config['table']) == {}
+
+def test_redis_runtime_database_check_table_object_exist(my_redis_client, rtdb_connect,
+                                                         data_config):
+    """Tests the check_table_object_exist method of the RedisDB class.
+
+    This test checks if the check_table_object_exist method can check whether a given object exists
+    in a given table on redis server successfully.
+
+    Args:
+        my_redis_client (Callable): Temporary redis client provided by pytest-redis's redisdb
+          fixture.
+        rtdb_connect (RuntimeDatabaseBase): Fixture for Redis runtime database.
+        data_config (dict): Fixture for test data configurations.
+
+    Returns:
+        None
+    """
+    my_redis_client.hset(data_config['table'], data_config['obj'], json.dumps(data_config['dict']))
+    assert rtdb_connect.check_table_object_exist(data_config['table'], data_config['obj']) is True
+
+def test_redis_runtime_database_del_table_object(my_redis_client, rtdb_connect, data_config):
+    """Tests the del_table_object method of the RedisDB class.
+
+    This test checks if the del_table_object method can delete an object from a given table
+    on redis server successfully.
+
+    Args:
+        my_redis_client (Callable): Temporary redis client provided by pytest-redis's redisdb
+          fixture.
+        rtdb_connect (RuntimeDatabaseBase): Fixture for Redis runtime database.
+        data_config (dict): Fixture for test data configurations.
+
+    Returns:
+        None
+    """
+    my_redis_client.hset(data_config['table'], data_config['obj'], json.dumps(data_config['dict']))
+    rtdb_connect.del_table_object(data_config['table'], data_config['obj'])
+    assert my_redis_client.hexists(data_config['table'], data_config['obj']) is False
+
+def test_invalid_table_param(my_redis_client, rtdb_connect, data_config):
+    """Tests the methods of the RedisDB class with invalid table param.
+
+    Args:
+        my_redis_client (Callable): Temporary redis client provided by pytest-redis's redisdb
+          fixture.
+        rtdb_connect (RuntimeDatabaseBase): Fixture for Redis runtime database.
+        data_config (dict): Fixture for test data configurations.
+
+    Returns:
+        None
+    """
+    data_config['table'] = None
```

Comment:
data_config seems to be used in many field, will this change influence other testing? Maybe use a copy of data_config.

Reply:
The default scope for `data_config` fixture is "function", so the change in some function will not influence other testing.